cpp/src/Slice/MetadataValidation.cpp

```diff
@@ -73,17 +74,15 @@ Slice::validateMetadata(const UnitPtr& p, string_view prefix, map<string, Metada
 
     // "amd"
     MetadataInfo amdInfo = {
-        {typeid(InterfaceDef), typeid(Operation)},
+        {typeid(InterfaceDecl), typeid(Operation)},
```

Comment:
Now that all metadata is on the `Decl` types, these are the types that should be passed into the validator.
I added a note in the header file mentioning this fact.